openpype/hosts/max/plugins/publish/extract_thumbnail.py

```diff
@@ -0,0 +1,80 @@
+import os
+import tempfile
+import pyblish.api
+from openpype.pipeline import publish
+from pymxs import runtime as rt
+
+
+class ExtractThumbnail(publish.Extractor):
+    """
+    Extract Thumbnail for Review
+    """
+
+    order = pyblish.api.ExtractorOrder
+    label = "Extract Thumbnail"
+    hosts = ["max"]
+    families = ["review"]
+
+    def process(self, instance):
+        # TODO: Create temp directory for thumbnail
+        # - this is to avoid "override" of source file
+        tmp_staging = tempfile.mkdtemp(prefix="pyblish_tmp_")
+        self.log.debug(
+            f"Create temp directory {tmp_staging} for thumbnail"
+        )
+        fps = int(instance.data["fps"])
+        instance.context.data["cleanupFullPaths"].append(tmp_staging)
+        filename = "{name}_thumbnail..jpg".format(**instance.data)
+        filepath = os.path.join(tmp_staging, filename)
+        filepath = filepath.replace("\\", "/")
+        thumbnail = self.get_filename(instance.name)
+
+        self.log.debug(
+            "Writing Thumbnail to"
+            " '%s' to '%s'" % (filename, tmp_staging))
+
+        preview_arg = self.set_preview_arg(
+            instance, filepath, fps)
+        rt.execute(preview_arg)
+
+        representation = {
+            "name": "thumbnail",
+            "ext": "jpg",
+            "files": thumbnail,
+            "stagingDir": tmp_staging,
+            "thumbnail": True
+        }
+
+        self.log.debug(f"{representation}")
+
+        if "representations" not in instance.data:
+            instance.data["representations"] = []
+        instance.data["representations"].append(representation)
+
+    def get_filename(self, filename):
+        return f"{filename}_thumbnail.0001.jpg"
+
+    def set_preview_arg(self, instance, filepath, fps):
+        job_args = list()
+        default_option = f'CreatePreview filename:"{filepath}"'
+        job_args.append(default_option)
+        frame_option = f"outputAVI:false start:1 end:1 fps:{fps}" # noqa
```

Comment:
Is it a good idea to always use frame 1. Wouldn't we rather use the start frame of the instance if it has it or the start frame of the timeline or alike?

Reply:
I meant more that if a thumbnail is generated - which happens only for "review" which has frame ranges set then why hardcode it to use frame 1 instead of the first frame of the review. 

What if there's tons of stuff going on outside of my frame range (e.g. some handles type of stuff, I wouldn't expect the thumbnail to show the handles but show at least a frame from within the frame range).

Reply:
I will use instance.data["framestart"]  as the first and end of the frame in the next update.